geomstats/riemannian_metric.py

```diff
@@ -176,17 +178,21 @@ def mean(self, points,
 
             mean_next = self.exp(tangent_vec=tangent_mean, base_point=mean)
 
-            sq_dist = self.squared_distance(mean_next, mean)
+            sq_dist = self.squared_dist(mean_next, mean)
             sq_dists_between_iterates.append(sq_dist)
 
-            variance = self.variance(mean_next, points, weights)
-
-            if sq_dist < epsilon * variance:
+            variance = self.variance(points=points,
+                                     weights=weights,
+                                     base_point=mean_next)
+            if sq_dist <= epsilon * variance:
                 break
 
             if it == n_max_iterations:
```

Comment:
please be consistent accross your code for comparing integers. Here you're using == and sometimes you're using is not